src/main/java/com/programmers/java/Calculator.java

```diff
@@ -0,0 +1,99 @@
+        Stack<BigDecimal> numberStack = new Stack<>();
+        while (index < postExp.length()) {
+            char ch = postExp.charAt(index);
+            if (Character.isDigit(ch)) {
+                numberStack.push(new BigDecimal(tokenizer(postExp)));
+            } else if (isOperator(ch)) {
+                numberStack.push(performOperation(ch, numberStack.pop(), numberStack.pop()));
+            }
+            index++;
+        }
+        return numberStack.pop();
+    }
+
+    private static String toPostfix(String expression) throws Exception {
+        Stack<Character> operatorStack = new Stack<>();
+        StringBuilder intermediateBuilder = new StringBuilder();
+        while(index < expression.length()) {
+            char ch = expression.charAt(index);
+            if(Character.isDigit(ch)) {
+                intermediateBuilder.append(tokenizer(expression)+ " ");
+            } else if (isOperator(ch)) {
+                while(!operatorStack.isEmpty() && getPriority(operatorStack.peek()) >= getPriority(ch)) {
+                    intermediateBuilder.append(operatorStack.pop());
+                }
+                operatorStack.push(ch);
+            } else if( expression.charAt(index) == '(') {
+                operatorStack.push(ch);
+            } else if( expression.charAt(index) == ')') {
+                while (operatorStack.peek() != '(') {
+                    intermediateBuilder.append(operatorStack.pop());
+                }
+                operatorStack.pop();
+            } else {
+                throw new Exception();
+            }
+
+            index++;
+        }
+        while (!operatorStack.isEmpty()) {
+            intermediateBuilder.append(operatorStack.pop());
+        }
+        return intermediateBuilder.toString();
+    }
+
+    private static String tokenizer(String exp) {
+        StringBuilder decimalBuilder = new StringBuilder();
+        while (index < exp.length() && (Character.isDigit(exp.charAt(index)) || exp.charAt(index) == '.')) {
+            decimalBuilder.append(exp.charAt(index));
+            index++;
+        }
+        index--;
+        return decimalBuilder.toString();
+    }
+    private static BigDecimal performOperation(Character operator, BigDecimal popLast, BigDecimal popSecondLast) {
+        switch (operator) {
+            case '+':
+                return popSecondLast.add(popLast);
+            case '-':
+                return popSecondLast.subtract(popLast);
+            case '*':
+                return popSecondLast.multiply(popLast);
+            case '/':
+                return popSecondLast.divide(popLast);
+        }
+        return BigDecimal.ZERO;
+    }
+
+    public static boolean isOperator(char ch) {
+        return (ch == '+' || ch == '-' || ch == '*' || ch == '/');
+    }
+
+    public static int getPriority(char ch) {
+        return (ch == '+' || ch == '-') ? 1 : (ch == '*' || ch == '/') ? 2 : 0;

```

Comment:
삼항연산자를 중첩하면 코드 가독성이 매우 안좋아집니다.
early return을 사용해서 간결하고 읽기 좋은 코드를 만들어줄 수 있습니다.

```suggestion
        if(ch == '+' || ch == '-')
            return 1;
        if(ch == '*' || ch == '/')
            return 2;
        return 0;
```

코딩테스트가 아니니, 한줄로 표현하는것보다 읽기 쉬운 코드를 짜는 방법을 익혀보아요~

Reply:
else를 중첩해서 가독성을 해치는 것 대신 return으로 간결하게 끝내는 것을 early return이라는 용어로 부르는군요ㅎㅎ 적용해보겠습니다!

Reply:
중괄호도 가독성에 도움되니 잊지 마시구요~